SSZipArchive/SSZipArchive.m

```diff
@@ -844,6 +845,135 @@ + (BOOL)createZipFileAtPath:(NSString *)path
+                }
+                //is symlink
+                BOOL isSymlink = NO;
+                if (mz_os_is_symlink(fullFilePath.fileSystemRepresentation) == MZ_OK)
+                    isSymlink = YES;
+                BOOL isDir;
+                [fileManager fileExistsAtPath:fullFilePath isDirectory:&isDir];
+                if (!isDir || isSymlink) {
+                    // file or symlink
+                    if (!isSymlink) {
+                        success &= [zipArchive writeFileAtPath:fullFilePath withFileName:fileName compressionLevel:compressionLevel password:password AES:aes];
+                    } else {
+                        success &= [zipArchive writeSymlinkFileAtPath:fullFilePath withFileName:fileName compressionLevel:compressionLevel password:password AES:aes];
+                    }                  
+                } else {
+                    // directory
+                    if (![fileManager enumeratorAtPath:fullFilePath].nextObject) {
+                        // empty directory
+                        success &= [zipArchive writeFolderAtPath:fullFilePath withFolderName:fileName withPassword:password];
+                    }
+                }
+                if (progressHandler) {
+                    complete++;
+                    progressHandler(complete, total);
+                }
+            }
+            success &= [zipArchive close];
+        }
+        return success;
+    }    
+}
+
+- (BOOL)writeSymlinkFileAtPath:(NSString *)path withFileName:(nullable NSString *)fileName compressionLevel:(int)compressionLevel password:(nullable NSString *)password AES:(BOOL)aes
+{
+    NSAssert((_zip != NULL), @"Attempting to write to an archive which was never opened");
+    //read symlink
+    char link_path[1024];
+    int32_t err = MZ_OK;
+    err = mz_os_read_symlink(path.fileSystemRepresentation, link_path, sizeof(link_path));
+    if (err != MZ_OK) {
+        NSLog(@"[SSZipArchive] Failed to read sylink");
+        return NO;
+    }
+
+    if (!fileName) {
+        fileName = path.lastPathComponent;
+    }
+    
+    zip_fileinfo zipInfo = {};    
+    [SSZipArchive zipInfo:&zipInfo setAttributesOfItemAtPath:path];
+    
+    //unpdate zipInfo.external_fa
+    uint32_t target_attrib = 0;
+    uint32_t src_attrib = 0;
+    uint32_t src_sys = 0;
+    mz_os_get_file_attribs(path.fileSystemRepresentation, &src_attrib);
+    src_sys = MZ_HOST_SYSTEM(MZ_VERSION_MADEBY);
+
+    if ((src_sys != MZ_HOST_SYSTEM_MSDOS) && (src_sys != MZ_HOST_SYSTEM_WINDOWS_NTFS)) {
+        /* High bytes are OS specific attributes, low byte is always DOS attributes */
+        if (mz_zip_attrib_convert(src_sys, src_attrib, MZ_HOST_SYSTEM_MSDOS, &target_attrib) == MZ_OK)
+            zipInfo.external_fa = target_attrib;
+        zipInfo.external_fa |= (src_attrib << 16);
+    } else {
+        zipInfo.external_fa = src_attrib;
+    }
+
+    uint16_t version_madeby = 3 << 8;//UNIX
```

Comment:
Only `3 << 8` is avaible to zip symlink. If it is changed to `19 << 8`, the symlink file will be zipped failed, which puzzles me.

Reply:
Did you ever solve this puzzle? I went ahead and merged in as everything looked good and tests passed.